backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeComponent.java

```diff
@@ -171,7 +167,13 @@ private Element doRender(VariableManager nodeVariableManager, String targetObjec
         }).collect(Collectors.toList());
         //@formatter:on
 
-        Size size = optionalPreviousNode.map(Node::getSize).orElseGet(() -> nodeSizeProvider.getSize(style, childNodes));
+        Size size;
+        if (childNodes.isEmpty()) {
+            size = nodeSizeProvider.getSize(style, childNodes);
+        } else {
+            size = optionalPreviousNode.map(Node::getSize).orElseGet(() -> nodeSizeProvider.getSize(style, childNodes));
+        }
+
```

Comment:
Why are you computing the size when there are no children but keeping the old size if there are children? The issue was that semantic changes cannot impact the size anymore, regardless if there are children or not.

Reply:
In its current state, the rendering cannot compute the size of an element according to its children.
This is to be solved by https://github.com/eclipse-sirius/sirius-components/pull/299
For now the rendering can only recompute sizes for childless nodes (and solve the issue with fans with conditional sizes) but rely on ELK (so, keep the precomputed size) for nodes with children.